demo/estimate_smplx.py

```diff
@@ -0,0 +1,295 @@
+    expression = np.array(expression)
+    pred_cams = np.array(pred_cams)
+    bboxes_xyxy = np.array(bboxes_xyxy)
+
+    # release GPU memory
+    del mesh_model
+    del extractor
+    torch.cuda.empty_cache()
+
+    # smooth
+    if args.smooth_type is not None:
+        smpl_poses = smooth_process(
+            smpl_poses.reshape(frame_num, -1, 21, 9),
+            smooth_type=args.smooth_type).reshape(frame_num, 21, 3, 3)
+        right_hand_pose = smooth_process(
+            right_hand_pose.reshape(frame_num, -1, 15, 9),
+            smooth_type=args.smooth_type).reshape(frame_num, 15, 3, 3)
+        left_hand_pose = smooth_process(
+            left_hand_pose.reshape(frame_num, -1, 15, 9),
+            smooth_type=args.smooth_type).reshape(frame_num, 15, 3, 3)
+        global_orient = smooth_process(
+            global_orient.reshape(frame_num, -1, 1, 9),
+            smooth_type=args.smooth_type).reshape(frame_num, 1, 3, 3)
+        jaw_pose = smooth_process(
+            jaw_pose.reshape(frame_num, -1, 1, 9),
+            smooth_type=args.smooth_type).reshape(frame_num, 1, 3, 3)
+        pred_cams = smooth_process(
+            pred_cams[:, np.newaxis],
+            smooth_type=args.smooth_type).reshape(frame_num, 3)
+
+    if smpl_poses.shape[1:] == (21, 3, 3):
+        smpl_poses = rotmat_to_aa(smpl_poses)
+        right_hand_pose = rotmat_to_aa(right_hand_pose)
+        left_hand_pose = rotmat_to_aa(left_hand_pose)
+        global_orient = rotmat_to_aa(global_orient)
+        jaw_pose = rotmat_to_aa(jaw_pose)
+    else:
+        raise Exception('Wrong shape of `smpl_pose`')
+    fullpose = np.concatenate((
+        global_orient.reshape(frame_num, 1, 3),
+        smpl_poses.reshape(frame_num, 21, 3),
+        jaw_pose.reshape(frame_num, 1, 3),
+        np.zeros(shape=(frame_num, 2, 3), dtype=global_orient.dtype),
+        left_hand_pose.reshape(frame_num, 15, 3),
+        right_hand_pose.reshape(frame_num, 15, 3),
+    ),
+                              axis=1)
+
+    if args.output is not None:
+        os.makedirs(args.output, exist_ok=True)
+        human_data = HumanData()
+        smplx = {}
+        smplx['fullpose'] = fullpose
+        smplx['betas'] = np.array(smpl_betas).reshape((-1, 10))
+        human_data['smplx'] = smplx
+        human_data['pred_cams'] = pred_cams
+        human_data.dump(osp.join(args.output, 'inference_result.npz'))
+
+    if args.show_path is not None:
+        frames_folder = osp.join(args.show_path, 'images')
+        os.makedirs(frames_folder, exist_ok=True)
+        array_to_images(
+            np.array(frames_iter)[frame_id_list], output_folder=frames_folder)
+
+        body_model = dict(
+            type='smplx',
+            num_betas=10,
+            use_face_contour=True,
+            use_pca=False,
+            flat_hand_mean=True,
+            model_path=args.body_model_dir,
+            keypoint_src='smplx',
+            keypoint_dst='smplx',
+        )
```

Comment:
Have they already been defined somewhere in input config?

Reply:
There is no definition of the `body_model `in the `configs/expose/expose.py`